AiTableTopGameMaster.ConsoleApp/Clients/ConsoleChatClient.cs

```diff
@@ -48,16 +50,53 @@ public async Task<ChatHistory> ChatAsync(ChatHistory history, CancellationToken
         
         try
         {
-            // Use the agent framework properly for chatting
-            await foreach (AgentResponseItem<ChatMessageContent> response in agent.InvokeAsync(history, cancellationToken: cancellationToken))
+            const int maxRevisionAttempts = 3;
+            int revisionAttempt = 0;
+            
+            while (revisionAttempt < maxRevisionAttempts)
             {
-                ChatMessageContent message = response.Message;
-                history.Add(message);
+                // Generate response from the game master agent
+                string gameMasterResponse = await GenerateGameMasterResponseAsync(history, cancellationToken);
                 
-                log.LogInformation("{Agent}: {Content}", agent.Name, message.Content);
+                // Review the response
+                OutputReviewResult reviewResult = await outputReviewer.ReviewOutputAsync(gameMasterResponse, cancellationToken);
                 
-                console.Markup($"{DisplayHelpers.AI}{agent.Name}:[/] ");
-                console.MarkupLineInterpolated($"{message.Content}");
+                if (reviewResult.IsAcceptable)
+                {
+                    // Response is acceptable, add to history and display to player
+                    ChatMessageContent acceptedMessage = new(AuthorRole.Assistant, gameMasterResponse);
+                    history.Add(acceptedMessage);
+                    
+                    log.LogInformation("{Agent}: {Content}", agent.Name, gameMasterResponse);
+                    console.Markup($"{DisplayHelpers.AI}{agent.Name}:[/] ");
+                    console.MarkupLineInterpolated($"{gameMasterResponse}");
+                    console.WriteLine();
+                    return history;
+                }
+                
+                // Response needs revision
+                revisionAttempt++;
+                log.LogDebug("Game master response needs revision (attempt {Attempt}/{Max}): {Issues}", 
+                    revisionAttempt, maxRevisionAttempts, string.Join(", ", reviewResult.Issues));
+                
+                if (revisionAttempt >= maxRevisionAttempts)
+                {
+                    // Max attempts reached, display the response anyway but with a warning
+                    console.MarkupLine($"{DisplayHelpers.System}[yellow]Note: Response may not fully follow game master best practices.[/]");
```

Comment:
I'm not sure this is the best message for the user to see. Additionally, System should not be used here with yellow, since System has its own formatting and Spectre.Console will error on this line.